tests/test_server.py

```diff
@@ -17,117 +41,284 @@ def get_app(self):
         )
-
-        self.io_loop.call_later(
-            1,
-            lambda: self.http_client.fetch(
-                self.get_url('/games/2'),
-                callback,
-            )
+        expected = {
+            'author': LOGIN,
+            'auto_restart': 300,
+            'auto_start': 300,
+            'board_size': 5,
+            'max_players': 15,
+            'results': None,
+            'title': data['title'],
+            'with_bot': False
+        }
+        self.assertDictEqual(expected, save_data)
+
+        self.assertEqual(len(server.game_rooms), 1)
+        deleted = yield self.client.fetch(
+            self.get_url('/games/{}?token={}'.format(ID, TOKEN)),
+            method='DELETE',
         )
+        self.assertEqual(deleted.code, 200)
+        self.assertEqual(len(server.game_rooms), 0)
+
+        args, kwargs = remove_method.call_args
+        expected = ({'_id': ID},)
+        self.assertEqual(expected, args)
+
+    @unittest.mock.patch(
+        'server.game_rooms', {
+            1: GameRoom(author=LOGIN, title='duplicated title'),
+            2: GameRoom(author=LOGIN),
+            3: GameRoom(author=LOGIN),
+            4: GameRoom(author=LOGIN)
+        }
+    )
+    @unittest.mock.patch(
+        'server.User.collection.find_one',
+        return_value=future_wrap({
+            '_id': ID,
+            'login': LOGIN,
+            'token': TOKEN,
+            'name': 'STXNext',
+        })
+    )
+    @tornado.testing.gen_test(timeout=100000)
+    def test_new_game_exceptions(self, user_get):
+
+        def new_room(body):
+            return self.client.fetch(
+                self.get_url('/games?token={}'.format(TOKEN)),
+                method='POST',
+                body=json.dumps(body),
+            )
 
-        self.wait()
-
-    def test_admin(self):
-        with unittest.mock.patch('server.User', autospec=True) as user:
-            user.configure_mock(**{
-                'get.return_value': user,
-                'admin': False,
-            })
-
-            response = self.fetch('/games/2', method='PUT', body="")
-
-            self.assertEqual(response.code, http.client.FORBIDDEN)
-
-        with unittest.mock.patch('server.User', autospec=True) as user:
-            user.configure_mock(**{
-                'get.return_value': user,
-                'admin': True,
+        bad_params = [
+            {},  # empty post data
+            {'title': 'too large', 'board_size': 15},
+            {'title': 'too small', 'board_size': 1},
+            {'title': 'wrong type', 'board_size': '1'},
+            {'title': 't'},  # too short
+            {'title': 't'.join([str(x) for x in range(0, 101)])},  # too long
+            {'title': 'duplicated title'},
+        ]
+
+        for params in bad_params:
+            with self.assertRaises(tornado.httpclient.HTTPError):
+                response = yield new_room(params)
+                self.assertEqual(response.code, 404)
```

Comment:
It's ok to have asserts in a loop, but if you will have fail you will not be able to see on which sample it failed. To check it you will have to use pdb.
Solution here is to use `self.assertEqual(response.code, 404, msg=...)` and in `msg` add something that will help (e.g. iteration from `enumerate(bad_params)` or title of the room).